openfl-tutorials/interactive_api/Director_Pytorch_TinyImageNet/envoy/base.py

```diff
@@ -0,0 +1,39 @@
+# Copyright (C) 2020-2021 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+"""Base shard descriptor."""
+
+
+class Dataset:
+    """Dataset class."""
+
+    def __len__(self):
+        """Return the len of the dataset."""
+        raise NotImplementedError
+
+    def __getitem__(self, index):
+        """Return an item by the index."""
+        raise NotImplementedError
+
+
+class ShardDescriptor:
```

Comment:
We have the base class for Shard Descriptor in the core package

Reply:
We have another base class in the core package. And that base class can replaced a core class in core package.